apps/costcontrol/js/settings/limitdialog.js

```diff
@@ -103,6 +107,11 @@ function dataLimitConfigurer(guiWidget, settings, viewManager, widgetRoot) {
             DATA_LIMIT_NUMERIC_FORMAT.test(numberDataLimit));
   }
 
+  function wouldBeValid(lowLimitValue) {
```

Comment:
nit: move functions to make them more aware about their dependencies. Use this order:
  # `isAlloweValue()`
  # `isValidValue()`
  # `wouldBeValid()`

And put all of them above `preventBadInput()`.